x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/LifecycleSettings.java

```diff
@@ -47,6 +48,13 @@
     public static final Setting<TimeValue> LIFECYCLE_STEP_MASTER_TIMEOUT_SETTING =
         Setting.positiveTimeSetting(LIFECYCLE_STEP_MASTER_TIMEOUT, TimeValue.timeValueSeconds(30), Setting.Property.Dynamic,
             Setting.Property.NodeScope);
+    // This setting configures how much time since step_time should ILM wait for a condition to be met. After the threshold wait time has
+    // elapsed ILM will likely stop waiting and go to the next step.
+    // Also see {@link org.elasticsearch.xpack.core.ilm.ClusterStateWaitUntilThresholdStep}
+    public static final Setting<TimeValue> LIFECYCLE_STEP_WAIT_TIME_THRESHOLD_SETTING =
+        Setting.positiveTimeSetting(LIFECYCLE_STEP_WAIT_TIME_THRESHOLD, TimeValue.timeValueHours(12), Setting.Property.Dynamic,
```

Comment:
Should we enforce a minimum setting? It would be bad to be spinning forever because someone put 1 second as their wait time.

Reply:
Good point. Shall we set it to one hour minimum? (this will make it impossible to integration test - https://github.com/elastic/elasticsearch/pull/70107/files/db8a13789e40a0b7f64263a0c88a257b21eb206c#diff-00728663bc72dd25e9832acf55fd6c4dd690a34d3c9ef6e5408464d81ff1f785R309 - however, maybe the unit tests we have around the `ClusterStateWaitUntilThresholdStep` are enough)

Reply:
I managed to integration test the rewind even with a `1h` minimum value in https://github.com/elastic/elasticsearch/pull/70107/commits/73e363dec50e3773d90390f1e50582a55fa05b3b